sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs

```diff
@@ -0,0 +1,259 @@
+
+                typeof(Secret).GetProperty(nameof(secret.Value)).SetValue(secret, null);
+
+                _versions[secret.Id.ToString()] = secret;
+            }
+        }
+
+        public void Dispose()
+        {
+            var deleteResult = _client.DeleteAsync(SecretName);
+        }
+
+        [Test]
+        public async Task GetAllVersionsAsyncForEach()
+        {
+            int actVersionCount = 0;
+
+            await foreach (var secret in _client.GetAllVersionsAsync(SecretName))
+            {
+                Assert.True(_versions.TryGetValue(secret.Id.ToString(), out Secret exp));
+
+                AssertSecretsEqual(exp, secret);
+
+                actVersionCount++;
+            }
+
+            Assert.AreEqual(VersionCount, actVersionCount);
+        }
+
+        [Test]
+        public async Task ListVersionEnumeratorMoveNext()
+        {
+            int actVersionCount = 0;
+
+            var enumerator = _client.GetAllVersionsAsync(SecretName);
+
+            while (await enumerator.MoveNextAsync())
+            {
+                Assert.True(_versions.TryGetValue(enumerator.Current.Id.ToString(), out Secret exp));
+
+                AssertSecretsEqual(exp, enumerator.Current);
+
+                actVersionCount++;
+            }
+
+            Assert.AreEqual(VersionCount, actVersionCount);
+        }
+
+
+        [Test]
+        public async Task GetAllVersionsByPageAsyncForEach()
+        {
+            int actVersionCount = 0;
+
+            await foreach (Page<SecretBase> currentPage in _client.GetAllVersionsAsync(SecretName).ByPage())
+            {
+                for (int i = 0; i < currentPage.Items.Length; i++)
+                {
+                    Assert.True(_versions.TryGetValue(currentPage.Items[i].Id.ToString(), out Secret exp));
+
+                    AssertSecretsEqual(exp, currentPage.Items[i]);
+
+                    actVersionCount++;
+                }
+            }
+
+            Assert.AreEqual(VersionCount, actVersionCount);
+        }
+
+        [Test]
+        public async Task ListVersionByPageEnumeratorMoveNext()
+        {
+            int actVersionCount = 0;
+
+            var enumerator = _client.GetAllVersionsAsync(SecretName).ByPage();
+
+            while (await enumerator.MoveNextAsync())
+            {
+                Page<SecretBase> currentPage = enumerator.Current;
+
+                for (int i = 0; i < currentPage.Items.Length; i++)
+                {
+                    Assert.True(_versions.TryGetValue(currentPage.Items[i].Id.ToString(), out Secret exp));
+
+                    AssertSecretsEqual(exp, currentPage.Items[i]);
+
+                    actVersionCount++;
+                }
+            }
+
+            Assert.AreEqual(VersionCount, actVersionCount);
+        }
+    }
+
+    public class KeyVaultTestBase
+    {
+        private static Lazy<Uri> s_vaultUri = new Lazy<Uri>(() => { return new Uri(Environment.GetEnvironmentVariable("AZURE_KEYVAULT_URL")); });
```

Comment:
If we want to make these tests work against CI we will need to add that variable into the YML files.